### PR TITLE
added redis tls support

### PR DESCRIPTION
**Pull request by dominik-cnx**

### CODE REVIEW

**Review by Kusig — APPROVED** [review]

File: gateleen-playground/src/main/java/org/swisspush/gateleen/playground/Server.java [hunk 1]
```diff
@@ -181,13 +181,15 @@ public void start() {
...
         Integer redisPort = (Integer) props.get("redis.port");
+        Boolean redisEnableTls = (Boolean) props.get("redis.enableTls");
 
         props.put(ExpansionHandler.MAX_EXPANSION_LEVEL_HARD_PROPERTY, "100");
         props.put(ExpansionHandler.MAX_EXPANSION_LEVEL_SOFT_PROPERTY, "50");
 
         RunConfig.deployModules(vertx, Server.class, props, success -> {
             if (success) {
-                redisClient = new RedisClient(vertx, new RedisOptions().setConnectionString("redis://" + redisHost + ":" + redisPort));
+                String protocol = redisEnableTls ? "rediss://" : "redis://";
```
Comment (hiddenalpha) [inline gateleen-playground/src/main/java/org/swisspush/gateleen/playground/Server.java:191]: WARN: if property is not set, we will run into an NPE here.

File: gateleen-runconfig/src/main/java/org/swisspush/gateleen/runconfig/RunConfig.java [hunk 2]
```diff
@@ -480,7 +496,8 @@ public static void deployModules(final Vertx vertx, Class verticleClass, Map<Str
         final Logger log = LoggerFactory.getLogger(verticleClass);
         String redisHost = (String) props.get("redis.host");
         Integer redisPort = (Integer) props.get("redis.port");
-        log.info("deploying redis module with host:" + redisHost + " port:" + redisPort);
+        Boolean redisEnableTls = (Boolean) props.get("redis.enableTls");
+        log.info("deploying redis module with host:" + redisHost + " port:" + redisPort + " TLS: " + redisEnableTls);
```
Comment (hiddenalpha) [inline gateleen-runconfig/src/main/java/org/swisspush/gateleen/runconfig/RunConfig.java:500]: [slf4j parameterized logging](https://www.tutorialspoint.com/slf4j/slf4j_parameterized_logging.htm)

File: gateleen-test/src/test/java/org/swisspush/gateleen/AbstractTest.java [hunk 3]
```diff
@@ -135,13 +135,15 @@ public static void setupBeforeClass(TestContext context) {
...
         Integer redisPort = (Integer) props.get("redis.port");
+        Boolean redisEnableTls = (Boolean) props.get("redis.enableTls");
 
         props.put(ExpansionHandler.MAX_EXPANSION_LEVEL_HARD_PROPERTY, "100");
         props.put(ExpansionHandler.MAX_EXPANSION_LEVEL_SOFT_PROPERTY, "4");
 
         RunConfig.deployModules(vertx, AbstractTest.class, props, success -> {
             if (success) {
-                RedisClient redisClient = new RedisClient(vertx, new RedisOptions().setConnectionString("redis://" + redisHost + ":" + redisPort));
+                String protocol = redisEnableTls ? "rediss://" : "redis://";
```
Comment (hiddenalpha) [inline gateleen-test/src/test/java/org/swisspush/gateleen/AbstractTest.java:145]: Same as [r1281747180](https://github.com/swisspost/gateleen/pull/522/files#r1281747180).

File: gateleen-playground/src/main/java/org/swisspush/gateleen/playground/Server.java (hunk 1 above)
Comment (ZhengXinCN) [reply gateleen-playground/src/main/java/org/swisspush/gateleen/playground/Server.java:191]: check added when read property

File: gateleen-test/src/test/java/org/swisspush/gateleen/AbstractTest.java (hunk 3 above)
Comment (ZhengXinCN) [reply gateleen-test/src/test/java/org/swisspush/gateleen/AbstractTest.java:145]: check added when read property

**Pull request merged**